drivers/ethernet/eth_enc28j60.c

```diff
@@ -335,19 +443,42 @@ static int eth_enc28j60_init(struct device *dev)
 {
 	const struct eth_enc28j60_config *config = dev->config->config_info;
 	struct eth_enc28j60_runtime *context = dev->driver_data;
+#ifdef SPI_LEGACY_API
 	struct spi_config spi_cfg;
-
+#else
+	/* SPI config */
+	context->spi_cfg.operation = SPI_WORD_SET(8);
+	context->spi_cfg.frequency = config->spi_freq;
+	context->spi_cfg.slave = config->spi_slave;
+	context->spi_cfg.vendor = 1;
+	context->spi_cfg.cs = &enc_spi_cs;
+	enc_spi_cs.gpio_pin = config->spi_cs_pin;
+
```

Comment:
this will need no handle the case when config->spi_cs_port is just "" (aka: undefined), which means there is no need of enc_spi_cs

Reply:
will do